opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -1117,8 +1130,18 @@ namespace Opm
         virtual void shutWell(int well_index) override {
             WellState::shutWell(well_index);
             const int np = numPhases();
-            for (int p = 0; p < np; ++p)
-                this->well_reservoir_rates_[np * well_index + p] = 0;
+
+            auto* resv = &this->well_reservoir_rates_[np*well_index + 0];
+            auto* wpi  = &this->productivity_index_[np*well_index + 0];
+
+            for (int p = 0; p < np; ++p) {
+                resv[p] = wpi[p] = 0.0;
```

Comment:
Easy to miss that you are assigning two arrays, please use one line for each.

Reply:
I've pushed an update to do this.